examples/herokilled/src/main.rs

```diff
@@ -61,6 +103,24 @@ fn main() -> Result<()> {
 
     let file = File::open(filepath.unwrap())?;
     let buf_reader = BufReader::new(file);
-    let mut parser = Parser::from_reader_with_visitor(buf_reader, MyVisitor)?;
-    parser.run_to_end()
+    let state = State::default();
+    let mut visitor = HandlerVisitor::with_state(state)
+        .with(
+            protos::CitadelUserMessageIds::KEUserMsgHeroKilled as u32,
+            hero_killed,
+        )
+        .with(protos::CitadelUserMessageIds::KEUserMsgChatMsg as u32, chat);
```

Comment:
i wonder if there's a way to associate message ids with message payloads automatically / semi-automatically. something to think about..
though this type of work would be out of scope of this pr.